tests/std/tests/GH_004109_iter_value_t_direct_initialization/test.compile.pass.cpp

```diff
@@ -0,0 +1,700 @@
+    (void) ranges::search(nil, nil, nil, nil, pred2);
+
+    (void) ranges::search_n(nil, nil, zero, val);
+    (void) ranges::search_n(nil, nil, zero, val, pred2);
+
+#if _HAS_CXX23
+    (void) ranges::contains(nil, nil, val);
+    (void) ranges::contains_subrange(nil, nil, nil, nil);
+
+    (void) ranges::starts_with(nil, nil, nil, nil);
+    (void) ranges::starts_with(nil, nil, nil, nil, pred2);
+
+    (void) ranges::ends_with(nil, nil, nil, nil);
+    (void) ranges::ends_with(nil, nil, nil, nil, pred2);
+#endif // _HAS_CXX23
+
+    // Modifying sequence operations
+    ranges::copy(nil, nil, nil);
+    ranges::copy_if(nil, nil, nil, pred);
+
+    ranges::copy_n(nil, zero, nil);
+
+    ranges::copy_backward(nil, nil, nil);
+
+    ranges::move(nil, nil, nil);
+
+    ranges::move_backward(nil, nil, nil);
+
+    ranges::fill(nil, nil, val);
+
+    ranges::fill_n(nil, zero, val);
+
+    ranges::transform(nil, nil, nil, unop);
+    ranges::transform(nil, nil, nil, nil, nil, binop);
+
+    ranges::generate(nil, nil, gen);
+
+    ranges::generate_n(nil, zero, gen);
+
+    (void) ranges::remove(nil, nil, val);
+    (void) ranges::remove_if(nil, nil, pred);
+
+    ranges::remove_copy(nil, nil, nil, val);
+    ranges::remove_copy_if(nil, nil, nil, pred);
+
+    ranges::replace(nil, nil, val, val);
+    ranges::replace_if(nil, nil, pred, val);
+
+    ranges::replace_copy(nil, nil, nil, val, val);
+    ranges::replace_copy_if(nil, nil, nil, pred, val);
+
+    ranges::swap_ranges(nil, nil, nil, nil);
+
+    ranges::reverse(nil, nil);
+    ranges::reverse_copy(nil, nil, nil);
+
+    ranges::rotate(nil, nil, nil);
+    ranges::rotate_copy(nil, nil, nil, nil);
+
+    ranges::shuffle(nil, nil, mt19937(1729));
+
+#if _HAS_CXX23
+    ranges::shift_left(nil, nil, 1729);
+    ranges::shift_right(nil, nil, 1729);
+#endif // _HAS_CXX23
+
+    ranges::sample(nil, nil, nil, zero, mt19937(1729));
+
+    (void) ranges::unique(nil, nil);
+    (void) ranges::unique(nil, nil, pred2);
+
+    ranges::unique_copy(nil, nil, nil);
+    ranges::unique_copy(nil, nil, nil, pred2);
+
+    // Partitioning operations
+    (void) ranges::is_partitioned(nil, nil, pred);
+    ranges::partition(nil, nil, pred);
+    ranges::partition_copy(nil, nil, nil, nil, pred);
+    ranges::stable_partition(nil, nil, pred);
+    (void) ranges::partition_point(nil, nil, pred);
+
+    // Sorting operations
+    (void) ranges::is_sorted(nil, nil);
+    (void) ranges::is_sorted(nil, nil, comp);
+
+    (void) ranges::is_sorted_until(nil, nil);
+    (void) ranges::is_sorted_until(nil, nil, comp);
+
+    ranges::sort(nil, nil);
+    ranges::sort(nil, nil, comp);
+
+    ranges::partial_sort(nil, nil, nil);
+    ranges::partial_sort(nil, nil, nil, comp);
+
+    ranges::partial_sort_copy(nil, nil, nil, nil);
+    ranges::partial_sort_copy(nil, nil, nil, nil, comp);
+
+    ranges::stable_sort(nil, nil);
+    ranges::stable_sort(nil, nil, comp);
+
+    ranges::nth_element(nil, nil, nil);

```

Comment:
`ranges::nth_element` takes an optional `comp`, and the other sorting algorithms are testing that.